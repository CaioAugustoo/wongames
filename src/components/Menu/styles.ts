import styled, { css } from 'styled-components'
import media from 'styled-media-query'

type MenuFullProps = {
  isOpen: boolean
}

export const Wrapper = styled.menu`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    justify-content: space-between;
    padding: ${theme.spacings.small} 0;

    position: relative;
  `}
`

export const LogoWrapper = styled.div`
  ${media.lessThan('medium')`
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  `}
`

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    cursor: pointer;
    color: ${theme.colors.white};
    width: 2.4rem;
    height: 2.4rem;
  `}
`

export const MenuGroup = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-grow: 1;
    justify-content: flex-end;

    > div {
      margin-left: ${theme.spacings.xsmall};
    }
  `}
`

export const MenuFull = styled.nav<MenuFullProps>`
  ${({ isOpen }) => css`
    opacity: ${isOpen ? 1 : 0};
  `}
`
