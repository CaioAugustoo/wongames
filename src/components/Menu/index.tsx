import { useState } from 'react'

import { Menu2 as MenuIcon } from '@styled-icons/remix-fill/Menu2'
import { ShoppingCart as ShoppingCartIcon } from '@styled-icons/material-outlined/ShoppingCart'
import { Search as SearchIcon } from '@styled-icons/material-outlined/Search'
import { Close as CloseIcon } from '@styled-icons/material-outlined/Close'

import Button from '../Button/index'
import Logo from '../Logo/index'
import * as S from './styles'

const Menu = () => {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <S.Wrapper>
      <S.IconWrapper onClick={() => setIsOpen(true)}>
        <MenuIcon aria-label="Open Menu" />
      </S.IconWrapper>
      <S.LogoWrapper>
        <Logo hideOnMobile />
      </S.LogoWrapper>
      <S.MenuGroup>
        <S.IconWrapper>
          <SearchIcon aria-label="Search" />
        </S.IconWrapper>
        <S.IconWrapper>
          <ShoppingCartIcon aria-label="Open Shopping Cart" />
        </S.IconWrapper>
      </S.MenuGroup>

      <S.MenuFull aria-hidden={!isOpen} isOpen={isOpen}>
        <CloseIcon aria-label="Close Menu" onClick={() => setIsOpen(false)} />
        <S.MenuNav>
          <S.MenuLink href="#">Home</S.MenuLink>
          <S.MenuLink href="#">Explore</S.MenuLink>
        </S.MenuNav>

        <S.RegisterBox>
          <Button fullWidth size="large">
            Log in now
          </Button>
          <span>or</span>

          <S.CreateAccount href="#" title="Sign Up">
            Sign Up
          </S.CreateAccount>
        </S.RegisterBox>
      </S.MenuFull>
    </S.Wrapper>
  )
}

export default Menu
