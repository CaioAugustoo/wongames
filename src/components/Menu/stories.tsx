import { Story, Meta } from '@storybook/react/types-6-0'
import Menu from '.'

export default {
  title: 'Menu',
  component: Menu
} as Meta

export const Default: Story = () => <Menu />

Default.parameters = {
  layout: 'fullscreen',
  backgrounds: {
    default: 'dark'
  }
}
