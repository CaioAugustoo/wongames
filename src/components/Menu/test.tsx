import { fireEvent, screen } from '@testing-library/dom'
import { renderWithTheme } from 'utils/tests/helpers'

import Menu from '.'

describe('<Menu />', () => {
  it('should render the menu', () => {
    const { container } = renderWithTheme(<Menu />)

    expect(screen.getByLabelText(/abrir menu/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/pesquisar/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/Carrinho de compras/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle the open/close mobile menu', () => {
    renderWithTheme(<Menu />)

    const fullMenuElement = screen.getByRole('navigation', { hidden: true })

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true')
    expect(fullMenuElement).toHaveStyle({ opacity: 0 })

    fireEvent.click(screen.getByLabelText(/abrir menu/i))
    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('false')
    expect(fullMenuElement).toHaveStyle({ opacity: 1 })

    fireEvent.click(screen.getByLabelText(/fechar menu/i))
    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true')
    expect(fullMenuElement).toHaveStyle({ opacity: 0 })
  })

  it('should show register box when logged out', () => {
    renderWithTheme(<Menu />)

    expect(screen.getAllByText(/entrar/i)).toHaveLength(2)
    expect(screen.getByText(/Registrar-se/i)).toBeInTheDocument()
  })

  it('should show wishlist and account when logged in', () => {
    renderWithTheme(<Menu username="caio" />)

    expect(screen.getByText(/Minha conta/i)).toBeInTheDocument()
    expect(screen.getByText(/Lista de desejos/i)).toBeInTheDocument()
    expect(screen.queryByText(/entrar/i)).not.toBeInTheDocument()
    expect(screen.queryByText(/registrar-se/i)).not.toBeInTheDocument()
  })
})
