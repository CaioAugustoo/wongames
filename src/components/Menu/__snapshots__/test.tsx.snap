// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render the menu 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c28 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c28:hover {
  background: #2499ae;
}

.c34 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c34:hover {
  background: #2499ae;
}

.c5 {
  color: #FAFAFA;
  width: 11rem;
  height: 3.3rem;
}

.c1 {
  display: none;
}

.c6 {
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 2.4rem 0;
  position: relative;
  z-index: 20;
}

.c2 {
  cursor: pointer;
  color: #FAFAFA;
  width: 2.4rem;
  height: 2.4rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 > div {
  margin-left: 1.6rem;
}

.c10 {
  color: #FAFAFA;
  position: relative;
  font-size: 1.6rem;
  margin: 0.3rem 2.4rem 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
}

.c10:hover::after {
  content: '';
  position: absolute;
  display: block;
  height: 0.3rem;
  background-color: #2499ae;
  -webkit-animation: hoverAnimation 0.2s forwards;
  animation: hoverAnimation 0.2s forwards;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #FAFAFA;
  position: fixed;
  z-index: 20;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  overflow: hidden;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
}

.c31 > svg {
  position: absolute;
  top: 0;
  right: 0;
  margin: 1.6rem;
  cursor: pointer;
  width: 2.4rem;
  height: 2.4rem;
}

.c31 .c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c31 .c9 {
  color: #030517;
  font-weight: 600;
  font-size: 2.0rem;
  margin-bottom: 2.4rem;
  -webkit-transform: translateY(3rem);
  -ms-transform: translateY(3rem);
  transform: translateY(3rem);
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

.c31 .c32 {
  -webkit-transform: translateY(3rem);
  -ms-transform: translateY(3rem);
  transform: translateY(3rem);
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 4.8rem 4.8rem;
}

.c33 > span {
  display: block;
  margin: 0.8rem 0;
  font-size: 1.2rem;
}

.c35 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2499ae;
  border-bottom: 0.2rem solid #2499ae;
}

.c13 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c13 .c16,
.c13 .c29 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

.c14 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
  z-index: 50;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
  z-index: 50;
}

.c17::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c30 {
  background: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
  cursor: initial;
}

.c15 {
  color: #FAFAFA;
  width: 2.4rem;
  height: 2.4rem;
  position: relative;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c23 {
  max-width: 100%;
}

.c25 {
  color: #47C8FF;
  font-size: 2.8rem;
}

.c27 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c19 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  cursor: initial;
}

.c19 .c20 {
  padding-bottom: 3.2rem;
}

.c19 .c22 {
  max-width: 20rem;
}

.c19 .c24 {
  font-size: 1.8rem;
}

.c19 .c26 {
  color: #030517;
  font-size: 1.6rem;
}

.c12 .c18 {
  width: 56rem;
}

@media (max-width:768px) {
  .c5 {
    width: 5.8rem;
    height: 4.8rem;
  }

  .c5 svg {
    height: 4.5rem;
    pointer-events: none;
  }

  .c5 .text {
    display: none;
  }
}

@media (max-width:768px) {
  .c1 {
    display: block;
  }
}

@media (min-width:768px) {
  .c6 {
    display: block;
  }
}

@media (max-width:768px) {
  .c4 {
    position: absolute;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media (min-width:768px) {
  .c8 {
    margin-left: 2.4rem;
  }
}

<menu
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        aria-label="Open Menu"
        class="c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 4h18v2H3V4zm0 7h12v2H3v-2zm0 7h18v2H3v-2z"
        />
      </svg>
    </div>
  </div>
  <div
    class="c4"
  >
    <a
      href="/"
    >
      <div
        class="c5"
        color="white"
      >
        <svg
          aria-label="Won Games"
          fill="none"
          role="img"
          viewBox="0 0 158 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M.0547602 15.2273l-.0354753 20.645C.0110436 40.6683 5.35471 43.5351 9.34604 40.8758L29.9416 27.1541l20.5483 13.7925c3.9822 2.6729 9.3356-.1755 9.3439-4.9715l.0343-19.9972c.0136-7.88161-7.5487-13.56538-15.1158-11.36093l-.8732.2544c-9.6 2.79669-19.8112 2.70243-29.358-.27101C7.35072 2.36704.0676653 7.71714.0547602 15.2273z"
            fill="url(#paint0_linear_logo)"
          />
          <path
            d="M30.1635 27.1596l-6.0335-4.292c-.3646-.2594-.8603.054-.7822.4946l1.2931 7.2974 5.5226-3.5z"
            fill="#110F28"
            fill-opacity=".15"
          />
          <path
            d="M7.59186 7.10277C2.48169 9.8576 3.07342 12.1027 3.07343 14.6027M57.2743 26.1031v1M3.05249 18.1027v2M57.2744 30.1031v5"
            stroke="#fff"
            stroke-linecap="round"
            stroke-opacity=".4"
            stroke-width="1.5"
          />
          <path
            d="M13.0526 15.1027l.0001 6M10.0403 18.1027l6.0246.0001"
            stroke="#0A0D27"
            stroke-linecap="round"
            stroke-width="2"
          />
          <ellipse
            fill="#0A0D27"
            rx="1.50616"
            ry="1.5"
            transform="matrix(1 0 0 1 44.723 13.6597)"
          />
          <ellipse
            fill="#0A0D27"
            rx="1.50616"
            ry="1.5"
            transform="matrix(1 0 0 1 49.7435 17.6031)"
          />
          <ellipse
            fill="#0A0D27"
            rx="1.50616"
            ry="1.5"
            transform="matrix(1 0 0 1 44.723 21.603)"
          />
          <ellipse
            fill="#0A0D27"
            rx="1.50616"
            ry="1.5"
            transform="matrix(1 0 0 1 39.7025 17.603)"
          />
          <path
            class="text"
            d="M79.2628 12.6327h-.06c-.68.1-1.2.44-1.56 1.02-.22.38-.33.78-.33 1.2 0 .1.01.2.03.3v.03l1.38 9.33c.08.54.34.99.78 1.35.42.32.89.48 1.41.48.12 0 .24-.01.36-.03.66-.1 1.17-.42 1.53-.96.26-.36.39-.77.39-1.23 0-.08-.01-.17-.03-.27l-1.38-9.33c-.08-.56-.34-1.03-.78-1.41-.42-.34-.9-.51-1.44-.51-.1 0-.2.01-.3.03zm18.78-.03c-.54 0-1.02.18-1.44.54-.44.36-.7.82-.78 1.38l-1.41 9.51c-.02.08-.03.15-.03.21 0 .52.17.98.51 1.38.36.46.84.73 1.44.81.1.02.21.03.33.03.66 0 1.21-.24 1.65-.72.3-.32.49-.7.57-1.14v-.06l1.3802-9.36v-.03c.02-.1.03-.2.03-.3 0-.42-.12-.82-.3602-1.2-.36-.58-.87-.92-1.53-1.02h-.06c-.1-.02-.2-.03-.3-.03zm-9.24 8.04h-.06c-.08 0-.16.01-.24.03-.36.04-.69.16-.99.36-.16.1-.3.22-.42.36-.08.08-.21.22-.39.42l-2.19 2.7c-.18.2-.42.48-.72.84-.28.34-.51.59-.69.75-.38.34-.75.58-1.11.72-.2.08-.41.14-.63.18-.14.02-.29.03-.45.03-.38 0-.74-.07-1.08-.21-.32-.12-.6-.29-.84-.51l.75 5.13v.06c.02.04.03.09.03.15.04.18.09.35.15.51.14.3.35.56.63.78.18.16.38.28.6.36.24.08.49.12.75.12h.09c.16 0 .3-.01.42-.03.46-.1.85-.32 1.17-.66l.3-.3 4.89-6.03 4.74 5.82c.24.28.39.45.45.51.38.38.83.61 1.35.69h.33c.28 0 .55-.05.81-.15.42-.18.77-.46 1.05-.84.18-.28.3-.59.36-.93v-.06l.75-5.01c-.34.3-.73.5-1.17.6-.26.08-.52.12-.78.12-.14 0-.29-.01-.45-.03-.28-.04-.55-.12-.81-.24s-.59-.37-.99-.75c-.26-.26-.59-.65-.99-1.17l-2.52-3.06c-.14-.18-.28-.34-.42-.48-.2-.2-.39-.35-.57-.45-.3-.2-.64-.31-1.02-.33h-.09zm19.8062.3c.28-.26.6-.46.96-.6.34-.14.69-.21 1.05-.21.46 0 .9.11 1.32.33.48.24.87.58 1.17 1.02.36-1.28 1.07-2.33 2.13-3.15 1.06-.82 2.26-1.23 3.6-1.23 1.66 0 3.07.59 4.23 1.77 1.16 1.16 1.74 2.57 1.74 4.23 0 1.66-.58 3.08-1.74 4.26-1.16 1.16-2.57 1.74-4.23 1.74-1.34 0-2.55-.4-3.63-1.2-1.06-.82-1.77-1.88-2.13-3.18-.22.34-.5.63-.84.87-.5.3-1.04.45-1.62.45-.42 0-.83-.08-1.23-.24-.3-.14-.56-.32-.78-.54.5 2.42 1.71 4.42 3.63 6 1.92 1.56 4.12 2.34 6.6 2.34 1.9 0 3.65-.47 5.25-1.41 1.62-.94 2.89-2.21 3.81-3.81.94-1.62 1.41-3.38 1.41-5.28 0-1.9-.47-3.65-1.41-5.25-.92-1.62-2.19-2.9-3.81-3.84-1.6-.94-3.35-1.41-5.25-1.41-2.46 0-4.65.79-6.57 2.37-1.92 1.56-3.14 3.55-3.66 5.97zm2.01-.09c-.66 0-1.21.24-1.65.72-.24.24-.41.53-.51.87-.06.22-.09.45-.09.69 0 .22.03.43.09.63.08.28.21.53.39.75.44.56 1.03.84 1.77.84.72 0 1.3-.27 1.74-.81.22-.26.37-.54.45-.84.04-.2.06-.4.06-.6 0-.18-.02-.35-.06-.51-.08-.38-.24-.71-.48-.99-.46-.5-1.03-.75-1.71-.75zm29.649-8.25c-.6 0-1.12.21-1.56.63-.46.44-.69.98-.69 1.62v6.15c.24-.28.53-.5.87-.66.42-.2.87-.3 1.35-.3h.06c.46 0 .9.1 1.32.3.34.16.64.38.9.66.06.06.11.12.15.18l9.81 11.55c.44.58 1.03.87 1.77.87.66 0 1.22-.25 1.68-.75.38-.42.57-.92.57-1.5v-6.12c-.26.26-.56.47-.9.63-.42.2-.87.3-1.35.3-.56 0-1.08-.13-1.56-.39-.34-.2-.63-.45-.87-.75l-9.87-11.67c-.22-.24-.48-.43-.78-.57-.28-.12-.58-.18-.9-.18zm13.98 0c-.62 0-1.14.21-1.56.63-.46.44-.69.98-.69 1.62v8.7c0 .42.14.8.42 1.14.44.56 1.05.84 1.83.84.54 0 1.02-.15 1.44-.45.52-.4.79-.91.81-1.53v-8.7c-.02-.7-.29-1.27-.81-1.71-.42-.36-.9-.54-1.44-.54zm-14.01 8.07c-.6 0-1.11.19-1.53.57-.44.4-.67.88-.69 1.44v8.67c0 .5.15.95.45 1.35.46.6 1.06.9 1.8.9.64 0 1.18-.23 1.62-.69.42-.42.63-.94.63-1.56v-8.67c-.02-.8-.41-1.39-1.17-1.77-.32-.16-.67-.24-1.05-.24h-.06z"
            fill="currentColor"
          />
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="paint0_linear_logo"
              x1="29.1592"
              x2="30.1547"
              y1="-7.39708"
              y2="49.1031"
            >
              <stop
                stop-color="#47C8FF"
              />
              <stop
                offset="1"
                stop-color="#2B96D8"
              />
              <stop
                offset="1"
                stop-color="#47C8FF"
              />
            </lineargradient>
          </defs>
        </svg>
      </div>
    </a>
  </div>
  <div
    class="c6"
  >
    <div
      class="c7 c8"
    >
      <a
        class="c9 c10"
        href="/"
      >
        Home
      </a>
      <a
        class="c9 c10"
        href="/games"
      >
        Library
      </a>
    </div>
  </div>
  <div
    class="c11"
  >
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        aria-label="Search"
        class="c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      <div
        class="c6"
      >
        <main
          class="c12"
        >
          <div
            class="c13"
          >
            <div
              class="c14"
            >
              <div
                class="c15"
              >
                <svg
                  aria-hidden="true"
                  aria-label="Shopping cart"
                  class="c3"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="c16 c17"
            >
              <div
                class="c18 c19"
              >
                <main
                  class="c20 c21"
                >
                  <img
                    alt="A gamer playing video games on the couch"
                    class="c22 c23"
                    role="img"
                    src="/img/empty.svg"
                  />
                  <h2
                    class="c24 c25"
                  >
                    Your cart is empty
                  </h2>
                  <p
                    class="c26 c27"
                  >
                    Visit our library and explore amazing games and offers
                  </p>
                  <a
                    class="c28"
                    href="/"
                  >
                    <span>
                      Back to the store
                    </span>
                  </a>
                </main>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="c29 c30"
            />
          </div>
        </main>
      </div>
      <div
        class="c1"
      >
        <a
          href="/cart"
        >
          <div
            class="c15"
          >
            <svg
              aria-hidden="true"
              aria-label="Shopping cart"
              class="c3"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c6"
    >
      <a
        class="c28"
        href="/sign-in"
      >
        <span>
          Sign In
        </span>
      </a>
    </div>
  </div>
  <nav
    aria-hidden="true"
    class="c31"
  >
    <svg
      aria-hidden="true"
      aria-label="Close Menu"
      class="c3"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
      />
    </svg>
    <div
      class="c7 c8"
    >
      <a
        class="c9 c10"
        href="/"
      >
        Home
      </a>
      <a
        class="c9 c10"
        href="/games"
      >
        Library
      </a>
    </div>
    <div
      class="c32 c33"
    >
      <a
        class="c34"
        href="/sign-in"
      >
        <span>
          Sign In
        </span>
      </a>
      <span>
        or
      </span>
      <a
        class="c35"
        href="/sign-up"
        title="Sign Up"
      >
        Sign Up
      </a>
    </div>
  </nav>
</menu>
`;
